ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessObjectsFinder.scala

```diff
@@ -42,6 +42,26 @@ object ProcessObjectsFinder {
     allObjectIds.diff(usedObjectIds).sortCaseInsensitive
   }
 
+  def findComponents(processes: List[ProcessDetails], componentId: String): List[ProcessComponent] = {
+    processes.flatMap(processDetails => processDetails.json match {
```

Comment:
maybe we can use collect? e.g.
    processes.flatMap(_.json).flatMap(_.nodes).collect {
      case node: WithComponent ...
    }